File: csharp/src/ev1/ut3_strings/ejercicios/pendientes/E03_Longitud_Pendiente.cs
/*
OBJETIVO: Quita espacios con Trim a una cadena leida y muestra Length
          (paridad con Python E03 StripLen). Menu do-while, 0 = salir.

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

string opcion;

do
{
    Console.WriteLine("=== E03 Trim y longitud (pendiente) ===");
    Console.WriteLine("1. Intentar ejercicio");
    Console.WriteLine("2. Ver objetivo");
    Console.WriteLine("0. Salir");
    Console.Write("Elige una opcion: ");
    opcion = Console.ReadLine() ?? "0";
    Console.WriteLine();

    switch (opcion)
    {
        case "1":
            Console.Write("Escribe una cadena (puedes dejar espacios): ");
            string texto = Console.ReadLine() ?? "";

            Console.WriteLine("Instrucciones:");
            Console.WriteLine("- Muestra Length del texto original.");
            Console.WriteLine("- Aplica Trim() y muestra Length del resultado.");

            int longitudOriginal = 0;
            string limpio = "";
            int longitudLimpio = 0;

            // TODO: asigna longitudOriginal, limpio (Trim) y longitudLimpio.

            Console.WriteLine("Original: \"" + texto + "\" Length=" + longitudOriginal);
            Console.WriteLine("Trim:     \"" + limpio + "\" Length=" + longitudLimpio);
            break;

        case "2":
            Console.WriteLine("Objetivo: usar Trim y Length, como strip() y len() en Python.");
            break;

        case "0":
            Console.WriteLine("Saliendo del ejercicio.");
            break;

        default:
            Console.WriteLine("Opcion no valida.");
            break;
    }

    Console.WriteLine();
} while (opcion != "0");

File: cpp/src/ev1/ut3_strings/ejercicios/pendientes/E03_Longitud_Pendiente.cpp
/*
OBJETIVO: Quita espacios (equivalente a Trim) a una cadena leida y muestra size()
          (paridad con Python E03 StripLen). Menu do-while, 0 = salir.

Autor: Agustin. A. Marquez. Pina
Contacto: agu1406@outlook.es
Repositorio GitHub: https://github.com/Agu1406/ClasesParticulares
Sitio web: https://www.agustinmarquez.dev
*/

#include <iostream>
#include <string>
using namespace std;

int main() {
    string opcion;

    do {
        cout << "=== E03 Trim y longitud (pendiente) ===" << endl;
        cout << "1. Intentar ejercicio" << endl;
        cout << "2. Ver objetivo" << endl;
        cout << "0. Salir" << endl;
        cout << "Elige una opcion: ";
        getline(cin, opcion);
        cout << endl;

        if (opcion == "1") {
            cout << "Escribe una cadena (puedes dejar espacios): ";
            string texto;
            getline(cin, texto);

            cout << "Instrucciones:" << endl;
            cout << "- Muestra size() del texto original." << endl;
            cout << "- Quita espacios con find_first_not_of / find_last_not_of y substr." << endl;
            cout << "- (No hay Trim() en C++17). Muestra size() del resultado." << endl;

            int longitudOriginal = 0;
            string limpio = "";
            int longitudLimpio = 0;

            // TODO: asigna longitudOriginal, limpio (sin espacios extremos) y longitudLimpio.

            cout << "Original: \"" << texto << "\" size=" << longitudOriginal << endl;
            cout << "Trim:     \"" << limpio << "\" size=" << longitudLimpio << endl;
        } else if (opcion == "2") {
            cout << "Objetivo: recortar espacios y usar size(), como strip() y len() en Python." << endl;
        } else if (opcion == "0") {
            cout << "Saliendo del ejercicio." << endl;
        } else {
            cout << "Opcion no valida." << endl;
        }

        cout << endl;
    } while (opcion != "0");

    return 0;
}
